cpp/src/join/hash_join.cu

```diff
@@ -300,7 +300,7 @@ hash_join<Hasher>::hash_join(cudf::table_view const& build,
   // need to store off the owning structures for some of the views in _build
   _flattened_build_table = structs::detail::flatten_nested_columns(
     build, {}, {}, structs::detail::column_nullability::FORCE);
-  _build = _flattened_build_table;
+  _build = _flattened_build_table->flattened_columns();
```

Comment:
This looks like it changes the contents of the build table, and goes beyond the scope of changing a return type. Can you explain more about why this changed?

Is there an overload of `cudf::detail::build_join_hash_table` accepting a `cudf::structs::detail::flattened_table`? I'm a little unsure of the context here.

Reply:
`_build = _flattened_build_table;` means `_build = _flattened_build_table.operator table_view();`. This is because of implicit conversion. And `operator table_view()` is just alias of `flattened_columns()`.

Reply:
What causes you confused also has been pointed out here: https://github.com/rapidsai/cudf/pull/12878#discussion_r1126932626.

It seems that we should rename that function (`flattened_columns`) and also remove the implicit conversion soon.

Reply:
Yes. This is definitely unclear.

Reply:
Should we consider some or all of those changes in the scope of this PR?